### ISSUE TITLE
Support reference to other components

### ISSUE THREAD

**Issue by davesnx** [davesnx/styled-ppx#15]
Parse nested properties and be able to create styles dependant on other's components.

```re
module Children = [%styled "display: block;"];
module Parent = [%styled {|
  $Children.styled {
    color: red;
  }
|}];

`$Children.styled` should be available. 
(This would only work in static components, warn that styled isn't available, even the compiler will say it).

/* ... */
<Parent>
  <Children />
</Parent>
```

**Comment by EduardoRFS** [comment]
If we separated the static and the dynamic parts we could actually support dynamic components on that

**Comment by davesnx** [comment]
There're a few workarounds on this. The first that comes to mind is using an empty className and reference it. I will keep this issue open to document this case.

**Comment by davesnx** [comment]
That's implemented in 0.30.

```
let firstCell = [%cx {||}];

let contentCell = [%cx {|
  padding: 10px;

  & $(firstCell) {
    padding-left: 0;
  }
|}];
```
